dev/bots/analyze.dart

```diff
@@ -17,34 +20,57 @@ final String dart = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Pl
 final String pub = path.join(flutterRoot, 'bin', 'cache', 'dart-sdk', 'bin', Platform.isWindows ? 'pub.bat' : 'pub');
 final String pubCache = path.join(flutterRoot, '.pub-cache');
 
+class ExitException implements Exception {
+  ExitException(this.exitCode);
+
+  final int exitCode;
+
+  @alwaysThrows
+  void apply() {
+    io_internals.exit(exitCode);
+  }
+}
+
+void exit(int exitCode) {
```

Comment:
Maybe also add a comment here explaining why we do this dance around `exit`. Future reader will probably appreciate it.

Reply:
done